subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -23,27 +23,46 @@
 import org.gradle.api.artifacts.dsl.DependencyHandler;
 import org.gradle.api.internal.tasks.AbstractTaskDependency;
 import org.gradle.api.internal.tasks.TaskDependencyResolveContext;
+import org.gradle.api.internal.tasks.testing.filter.DefaultTestFilter;
+import org.gradle.api.internal.tasks.testing.junit.JUnitTestFramework;
+import org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestFramework;
 import org.gradle.api.model.ObjectFactory;
 import org.gradle.api.plugins.JavaPlugin;
 import org.gradle.api.plugins.JvmTestSuitePlugin;
 import org.gradle.api.plugins.jvm.ComponentDependencies;
-import org.gradle.api.plugins.jvm.JUnitPlatformTestingFramework;
-import org.gradle.api.plugins.jvm.JUnitTestingFramework;
 import org.gradle.api.plugins.jvm.JvmTestSuite;
 import org.gradle.api.plugins.jvm.JvmTestSuiteTarget;
-import org.gradle.api.plugins.jvm.JvmTestingFramework;
-import org.gradle.api.provider.ProviderFactory;
+import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.SourceSet;
 import org.gradle.api.tasks.SourceSetContainer;
 import org.gradle.api.tasks.TaskDependency;
 
+import javax.annotation.Nullable;
 import javax.inject.Inject;
 
 public abstract class DefaultJvmTestSuite implements JvmTestSuite {
+    private enum Frameworks {
+        JUNIT4, JUNIT_JUPITER, NONE;
+    }
+    private static class TestingFramework {
```

Comment:
By moving this to the `DefaultJvmTestSuite` class out of the `JvmTestSuite` interface, we're removing the possibility of the user doing something like "retrieve all test suites using JUnit Jupiter and configure them all in same way" - perhaps by adding a dependency or fixture they would all require.  Why hide this from the interface?

Reply:
For 7.3, I wanted to reduce the scope of the new APIs as much as we could. 

In the future, we can expose this again or in a different way depending on which use cases someone has. For the initial pass at this, I think we just want to cover:

- A build with some tests that use JUnit4
- A build with some tests that use JUnit Jupiter (aka JUnit 5)

If a build needs anything else, they won't use the test suite API just yet.